src/main/scala/com/microsoft/hyperspace/actions/NoChangesException.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.actions
+
+/**
+ * This exception represents a No-op required from Hyperspace. Use this exception when a
+ * hyperspace action is not necessary for index maintenance.
+ * For example, if the data source has not changed since the last time an index was created on it,
+ * we don't need to do anything when user calls a `refreshIndex()`.
+ *
+ * Hyperspace actions silently catch this exception and do not fail the application.
```

Comment:
I guess `Action.run` catches this, not the individual actions?

Reply:
oh so I wrote this because Action.run is actually called by its concrete children, so technically, it's the object of individual Action subclasses which run it. That was my thinking when I wrote this down. I will correct this.
Thanks @imback82

Reply:
thanks @imback82 , updated